trieste/acquisition/function.py

```diff
@@ -697,6 +697,113 @@ def ehvi_based_on_partitioned_cell(
+        """
+        :param sample_size: The number of samples for each batch of points.
+        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive, or
+            ``jitter`` is negative.
+        """
+        tf.debugging.assert_positive(sample_size)
+        tf.debugging.assert_greater_equal(jitter, 0.0)
+
+        super().__init__()
+
+        self._sample_size = sample_size
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+        return (
+            f"BatchMonteCarloHypervolumeExpectedImprovement({self._sample_size!r},"
+            f" jitter={self._jitter!r})"
+        )
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model over the specified ``dataset``. Must have event shape [1].
+        :return: The batch expected hypervolume improvement acquisition function.
+        """
+
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+
+        _pf = Pareto(mean)
+        _reference_pt = get_reference_point(_pf.front)
+
+        sampler = BatchReparametrizationSampler(self._sample_size, model)
+
+        return batch_ehvi(sampler, self._jitter, _pf, _reference_pt)
+
+
+def batch_ehvi(
+    sampler, sampler_jitter: float, pareto: Pareto, reference_point: TensorType
+) -> AcquisitionFunction:
+    """
+    :param sampler: The posterior sampler to sample the possible observations @ query points `at`
+    :param sampler_jitter: The size of the jitter to use in sampler
+    :param pareto: Pareto class
+    :param reference_point: The reference point for calculating hypervolume
+    :return: The batch expected hypervolume improvement acquisition
+        function for objective minimisation.
+    """
+
+    def acquisition(at: TensorType) -> TensorType:
+        _batch_size = at.shape[-2]  # B
+
+        def gen_q_subset_indices(q: int) -> list:  # generate all subsets of [1, ..., q] as indices
+            indices = list(range(q))
+            return [tf.constant(list(combinations(indices, i))) for i in range(1, q + 1)]
+
+        samples = sampler.sample(at, jitter=sampler_jitter)  # [..., S, B, num_obj]
+
+        q_subset_indices = gen_q_subset_indices(_batch_size)
+
+        hv_contrib = 0.0
+        lb_points, ub_points = pareto.hypercell_bounds(
+            tf.constant([-inf] * samples.shape[-1], dtype=at.dtype), reference_point
+        )
+
+        def hv_contrib_on_samples(obj_samples: TensorType) -> TensorType:
+            # [..., S, Cq_j, j, num_obj] -> [..., S, Cq_j, num_obj]
+            overlap_vertices = tf.reduce_max(obj_samples, axis=-2)
```

Comment:
Can you explain what an overlap vertices is?

Reply:
Added an explanation: `# calculate samples overlapped area's hvi for obj_samples`, overlap vertices is the vertices of the overlapped area, hope this has increased some interpretability.